specification/_types/Retriever.ts

```diff
@@ -39,13 +39,60 @@ export class RetrieverContainer {
   text_similarity_reranker?: TextSimilarityReranker
   /** A retriever that replaces the functionality of a rule query. */
   rule?: RuleRetriever
+  /** A retriever that re-scores only the results produced by its child retriever. */
+  rescorer?: RescorerRetriever
+  /** A retriever that supports the combination of different retrievers through a weighted linear combination. */
+  linear?: LinearRetriever
+  /**
+   * A pinned retriever applies pinned documents to the underlying retriever.
+   * This retriever will rewrite to a PinnedQueryBuilder.
+   */
+  pinned?: PinnedRetriever
 }
 
 export class RetrieverBase {
   /** Query to filter the documents that can match. */
   filter?: QueryContainer | QueryContainer[]
   /** Minimum _score for matching documents. Documents with a lower _score are not included in the top documents. */
   min_score?: float
+  /** Retriever name. */
+  _name?: string
+}
+
+export class RescorerRetriever extends RetrieverBase {
+  /** Inner retriever. */
+  retriever: RetrieverContainer
+  rescore: Rescore | Rescore[]
+}
+
+export class LinearRetriever extends RetrieverBase {
+  /** Inner retrievers. */
+  retrievers?: InnerRetriever[]
+  rank_window_size: integer
+}
+
+export class PinnedRetriever extends RetrieverBase {
+  /** Inner retriever. */
+  retriever: RetrieverContainer
+  ids?: string[]
+  docs?: SpecifiedDocument[]
+  rank_window_size: integer
+}
+
+export class InnerRetriever {
+  retriever: RetrieverContainer
+  weight: float
+  normalizer: ScoreNormalizer
+}
+
+export enum ScoreNormalizer {
+  none,
+  minmax
```

Comment:
can I ask to please add `l2_norm` here? 🥺 
it was just merged this week cc @mridula-s109

Reply:
Hi @ioanatia , we have to add the `l2_norm` in a separate PR since this one should as well be in 8.18.